### PR TITLE
GameStates - second attempt

### PR DESCRIPTION
**Pull request by totine**
Setters removed.
After running project, It is possible to move through all states by input correct statement (there are hints in output which ones).

States:
- **Initial state** - round parameters will be choosed here, move to Game In Progress;
- **Game In Progress** - the main state of round, it is possible to move to Draw and Victory state or to Final (if player want to quit);
- **Victory and Draw states** - the Final State is after them;
- **Final State** - round is over, asks about game continuation (move to Initial State) or exit (move to Terminate State);
- **Terminate State** - inform about the game is end.

Pom improved - project should be built correctly.
Code coverage - ca 80%.

GameState interface has methods for:
- display current state,
- ask for next input,
- consume input (and doing something with it),
- move to next state,
- check is game over.

Strings with information are collected in enums (I think it's to improve)
"Ifology" to impove too

### CODE REVIEW

File: src/main/java/akademia/ox/Main.java [hunk 1]
```diff
@@ -2,6 +2,13 @@
 
 public class Main {
     public static void main(String[] args) {
-        System.out.println("Tylko testy");
+        GameState state = new InitialState();
+        while (!state.isGameOver()) {
+            System.out.println(state.showStateInfo());
+            state = state.moveToNextState();
+            if (state.getClass().equals(GameInProgress.class)) {
+                ((GameInProgress) state).setVictory();
```
Comment (JaroslawZaczyk) [inline src/main/java/akademia/ox/Main.java]: Try to remove this casting.

File: Readme.md [hunk 2]
```diff
@@ -0,0 +1 @@
+Run from Main class
```
Comment (JaroslawZaczyk) [inline Readme.md:1]: Better way: To run this game in terminal use java -jar /jar's_name/

File: pom.xml [hunk 3]
```diff
@@ -14,7 +14,20 @@
         <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
     </properties>
     <build>
+<plugins>
+<plugin>
+    <artifactId>maven-jar-plugin</artifactId>
+    <configuration>
+    <archive>
+    <manifest>
```
Comment (JaroslawZaczyk) [inline pom.xml:22]: please reformat code :smiley:

**Review by JaroslawZaczyk — APPROVED** [review]
Project works. There was a few little mistakes, but generally looks fine. Good job. Let's go with the rest of your TicTacToe game features.

File: Readme.md (hunk 2 above)
Comment (totine) [reply Readme.md:1]: OK

**Pull request merged**